src/representation_theory/group_cohomology/resolution.lean

```diff
@@ -281,6 +281,38 @@ begin
     finsupp.llift_apply A k k],
 end
 
+/-- Auxiliary lemma for defining group cohomology, used to show that the isomorphism
+`diagonal_hom_equiv` commutes with the differentials in two complexes which compute
+group cohomology. -/
+lemma diagonal_hom_equiv_symm_partial_prod_succ
+  (f : (fin n → G) → A) (g : fin (n + 1) → G) (a : fin (n + 1)) :
+  ((diagonal_hom_equiv n A).symm f).hom (finsupp.single (fin.partial_prod g ∘ a.succ.succ_above) 1)
+    = f (fin.contract_nth a (*) g) :=
+begin
+  simp only [diagonal_hom_equiv_symm_apply, function.comp_app, fin.succ_succ_above_zero,
+    fin.partial_prod_zero, map_one, fin.coe_eq_cast_succ, fin.succ_succ_above_succ,
+    linear_map.one_apply, fin.partial_prod_succ],
+  congr,
+  ext,
+  rcases lt_trichotomy (x : ℕ) a with (h|h|h),
```

Comment:
What's the statement at this point? It looks like there might be a lemma that can be extracted about contract_nth and partial_prod

Reply:
Applied this and the other comments, thanks!